File: src/main/java/org/funivan/lologa/tile/Score/MaxScore.java
package org.funivan.lologa.tile.Score;

import org.funivan.lologa.tile.TileInterface;
import org.funivan.lologa.tiles.TilesInterface;

public class MaxScore implements ScoreInterface {
    private final Iterable<TileInterface> tiles;

    public MaxScore(Iterable<TileInterface> tiles) {
        this.tiles = tiles;
    }
    public MaxScore(TilesInterface tiles) {
        this.tiles = tiles.all();
    }

    @Override
    public int value() {
        int score = 0;
        for (TileInterface tile : this.tiles) {
            final int tileScore = tile.score().value();
            score = tileScore > score ? tileScore : score;
        }
        return score;
    }

    @Override
    public String toString() {
        return "Score{" + this.value() + '}';
    }
}

File: src/main/java/org/funivan/lologa/tile/Score/MaxScore.kt
package org.funivan.lologa.tile.Score

import org.funivan.lologa.tile.TileInterface
import org.funivan.lologa.tiles.TilesInterface

class MaxScore : ScoreInterface {
    private val tiles: Iterable<TileInterface>

    constructor(tiles: Iterable<TileInterface>) {
        this.tiles = tiles
    }

    constructor(tiles: TilesInterface) {
        this.tiles = tiles.all()
    }

    override fun value(): Int {
        var score = 0
        for (tile in this.tiles) {
            val tileScore = tile.score().value()
            score = if (tileScore > score) tileScore else score
        }
        return score
    }

    override fun toString(): String {
        return "Score{" + this.value() + '}'
    }
}
